qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -75,6 +75,10 @@ def run(self, dag):
         Raises:
             TranspilerError: if dag wider than self.coupling_map
         """
+        if self.coupling_map is None:
```

Comment:
Would this ever not be set by the initializer? Just curious why this assertion is here rather than at the end of the init.

Reply:
It's a weird quirk with the layout and routing passes. The way they get initialized in the preset pass managers is if there is not connectivity constraint they get initialized with `coupling_map=None` (or now a target without qubits) and then the output pass manager doesn't include the pass. If we raised an error at init the function building the preset pass managers would fail blocking a `transpile()` even though not having a connectivity constraint is valid. It's something I want to cleanup after #6403 so we don't waste time constructing passes we're not using but at least for right now it's necessary to not raise during `__init__`